### PR TITLE
add SettingsSection component

### PR DESCRIPTION
**Pull request by GretaD**

### CODE REVIEW

File: src/components/SettingsSection/SettingsSection.vue [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+</template>
+
+<script>
+export default {
+	name: 'SettingsSection',
+	components: {
+
+	},
+	props: {
+		initial: {
```
Comment (juliushaertl) [inline src/components/SettingsSection/SettingsSection.vue]: title and description are missing, initial is not needed

File: src/components/SettingsSection/SettingsSection.vue [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			type: Object,
+			required: true
+		}
+	}
+}
+
+</script>
+
+<style scoped>
+	p {
```
Comment (juliushaertl) [inline src/components/SettingsSection/SettingsSection.vue]: Styles for .section should be taken over from settings.scss

File: src/components/SettingsSection/SettingsSection.vue [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="section">
+		<h2>{{ title }}</h2>
```
Comment (juliushaertl) [inline src/components/SettingsSection/SettingsSection.vue]: We should also add the option to specify a documentation URL here that is then linked with an info icon like we use it at some places:

image

File: src/components/SettingsSection/SettingsSection.vue [hunk 4]
```diff
@@ -0,0 +1,75 @@
+<!--
+  - @copyright Copyright (c) 2019 Julius Härtl <jus@bitgrid.net>
```
Comment (ChristophWurst) [inline src/components/SettingsSection/SettingsSection.vue]: no? :)

File: src/components/SettingsSection/SettingsSection.vue [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="section">
+		<h2>{{ title }}</h2>
+		<p>{{ description }}</p>
+		<a href="docUrl" class="icon-info" :title="t('settings', 'Open Documentation')"/>
```
Comment (ChristophWurst) [inline src/components/SettingsSection/SettingsSection.vue]: test data to remove?

File: src/components/SettingsSection/SettingsSection.vue (hunk 4 above)
Comment (GretaD) [reply src/components/SettingsSection/SettingsSection.vue]: I adapted the SettingsSection from richdocuments that Julius did. So we are co-authors :) need to add mine too

File: src/components/SettingsSection/SettingsSection.vue [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+<script>
+export default {
+	name: 'SettingsSection',
+	components: {
+
+	},
+	props: {
+		title: {
+			type: String,
+			default: '',
```
Comment (ChristophWurst) [inline src/components/SettingsSection/SettingsSection.vue]: not necessary if the prop is required

**Comment by ChristophWurst** [comment]
Please rebase

File: src/components/SettingsSection/SettingsSection.vue [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="section">
+		<h2>{{ title }}</h2>
+		<p>{{ description }}</p>
+		<a href="docUrl" class="icon-info" :title="title" />
```
Comment (georgehrke) [inline src/components/SettingsSection/SettingsSection.vue]: Shouldn't `docUrl` be a property?

**Comment by skjnldsv** [comment]
Rebased and ready to go!

@GretaD see what I did, it might help you for the next pending pr: :wink: 
- Added documentation
- Imported icons so we don't rely on the server css
- Used different class names so we don't rely on the server css
- Used BEM class naming convention
- Made sure accessibility was ok (external doc title translation) 
- Used our existing variables so we don't have random px units laying around :grin: 

image

**Comment by skjnldsv** [comment]
Looks perfect in the documentation too :)

**Pull request merged**